src/components/shared/ButtonWithPanel/ClosePanelContext.js

```diff
@@ -0,0 +1,14 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+// This is a very simple context that's used for communication between the panel
+// and its content, so that it can be closed from inner buttons.
+
+import * as React from 'react';
+
+export const ClosePanelContext: React.Context<
+  (void) => mixed
+> = React.createContext(() => {});
```

Comment:
This may be slightly overengineered but I thought this could enable more usage of the ButtonWithPanel in the future. For example if we rework the Publish Panel in the future this could be useful too.

Reply:
Same thing ^